usaspending_api/common/helpers/etl_helpers.py

```diff
@@ -59,3 +60,19 @@ def update_c_to_d_linkages(type):
     logger.info('Count of unlinked %s records after updates: %s' % (type, str(ending_unlinked_count)))
 
     logger.info('Finished all queries in %s seconds' % str(datetime.now() - total_start))
+
+
+def upsert_transactions(ids, linkage_type):
+
+    if ids:
+        update_award_ids = tuple(set(ids))  # Convert to tuple and remove duplicates.
+        with timer("updating awards to reflect their latest associated transaction info", logger.info):
+            update_awards(update_award_ids)
+        with timer("updating award category variables", logger.info):
+            update_award_categories(update_award_ids)
+
+        with timer("updating C->D linkages", logger.info):
+            update_c_to_d_linkages(linkage_type)
```

Comment:
New comment unrelated to the others here... So before, `update_c_to_d_linkages` was called if there were any transaction or award changes.  Now it seems it's being called only for award changes.  Do we know if that's ok?  Looking through the SQL, it doesn't seem like they touch the transaction tables, so this might be fine just want to ensure this was intentional and tested.

Reply:
This was noted early on, but the change was more about wanting to keep code clean than desiring an actual business change, and it would probably be best to just move it back. As for testing, since my only way is to test by exhaustion, I've held off on doing a thorough test until the structure of these changes are roughly agreed upon on a style level.